content/committees/specification/tckprocess/index.md

```diff
@@ -44,6 +44,19 @@ TCK tests must not be packaged in the `jakarta.*` namespace.  At the
 current time TCKs may be packaged in any other namespace, however, the
 namespace format of `ee.jakarta.tck.<spec-name>` is recommended.
 
+## TCK types and requirements
+
+There are 3 different types of TCKs, detailed below.
+- Type 1: Tests that only apply when running outside of a platform (or profile), such as on Java SE, to test behavior defined by the specification for that mode of execution.
+- Type 2: Tests that only apply when running in a platform (or profile) to test behavior defined by the specification for that mode of execution.
+- Type 3: Tests that apply to the behavior defined by a specification for all modes of execution.
+
+The individual specification TCK must contain Type 2 or Type 3 tests while Type 1 tests are optional. A profile-ready mechanism must be provided for running the required TCK tests (types 2 and 3) in the profiles they are included in: platform, web profile or core profile. Type 1 tests must be excluded and not required to be executed.
+
```

Comment:
This paragraph as written needs some changes.

Is your goal to ensure that the EE 11 Platform Specification does not go to ballot until all Type 1 tests are extended by a Type 2 test to allow that test to run as a Type 2 test?  Or is it more helpful to require that all Platform TCK testing is done with a Profile/Platform TCK?

I think we need to allow Type 1 tests to be the only tests for all Specification waves as only the Type 1 tests can be passed without having an implementation of the higher Specification wave(s).

Including Type 2/3 tests in individual specification TCK introduces cycles between Specifications that cross the higher Specification waves.

Reply:
@scottmarlow The 2 types of tests are independent. Type 1 tests do not need to be extended by Type 2 tests. However, a spec must contain type 2/3 tests. This paragraph does not talk about cross spec tests. It is pretty much about how tests are executed whether it is under Java SE or Jakarta EE mode. Type 1 is Java SE tests while type 2/3 are Jakarta EE mode tests.